### PR TITLE
test: init container e2e tests

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
This PR adds some minimal end-to-end tests for init container resource computation in light of this upstream PR: https://github.com/kubernetes-sigs/karpenter/pull/569.

**How was this change tested?**
`/karpenter snapshot` + local `make e2etests`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jmdeal** [comment]
Blocked by some changes that need to be changed in the upstream PR to support specifying init container images: https://github.com/kubernetes-sigs/karpenter/pull/569#discussion_r1462624455

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

**Review by jmdeal — COMMENTED** [review]
/karpenter snapshot

File: test/suites/integration/scheduling_test.go [hunk 1]
```diff
@@ -70,353 +72,556 @@ var _ = Describe("Scheduling", Ordered, ContinueOnFailure, func() {
...
+				RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("350m")},
+				},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("300m")},
+			})),
+		)
+		DescribeTable(
+			"should provision a right-sized node when a pod has InitContainers (memory)",
```
Comment (jonathan-innis) [inline test/suites/integration/scheduling_test.go:510]: Do we need both CPU and memory testing? I'd assume that functional testing should catch that we are getting this wrong with different resources? We aren't testing with ephemeral storage and the cost of running additional testing in E2E is going to get more expensive over time

File: test/suites/integration/scheduling_test.go [hunk 2]
```diff
@@ -70,353 +72,556 @@ var _ = Describe("Scheduling", Ordered, ContinueOnFailure, func() {
...
+			})
+			pod := test.Pod()
+			env.ExpectCreated(pod, nodeClass, nodePoolLowPri, nodePoolHighPri)
+			env.EventuallyExpectHealthy(pod)
+			env.ExpectCreatedNodeCount("==", 1)
+			Expect(ptr.StringValue(env.GetInstance(pod.Spec.NodeName).InstanceType)).To(Equal("c5.large"))
+			Expect(env.GetNode(pod.Spec.NodeName).Labels[corev1beta1.NodePoolLabelKey]).To(Equal(nodePoolHighPri.Name))
+		})
+
+		DescribeTable(
```
Comment (jonathan-innis) [inline test/suites/integration/scheduling_test.go:430]: This is 6 tests. Maybe they're all necessary (I hadn't looked into them that deeply) but is there a way to cut it down to a smaller set of tests? If you think they are all necessary, I'm fine taking these changes in.

File: test/suites/integration/scheduling_test.go (hunk 1 above)
Comment (jmdeal) [reply test/suites/integration/scheduling_test.go:510]: Probably not no, I think it is valuable to include at least one test with multiple resource types but I think we can remove the memory `DescribeTable`.

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**